marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/hub-entities.xqy

```diff
@@ -255,14 +255,13 @@ declare %private function hent:build-sort-operator(
 {
   let $states :=
     for $path-expression in map:keys($sortable-properties)
-    let $state-name-prefix :=
-      let $sort-info := map:get($sortable-properties, $path-expression)
-      return fn:concat(map:get($sort-info, "entity-title"), "_", map:get($sort-info, "property-name"))
-
+    let $sort-info := map:get($sortable-properties, $path-expression)
+    let $propert-data-type := hent:get-indexable-datatype(map:get($sort-info, "property-data-type"))
```

Comment:
Got a typo here - should be "property" - but let's call this "indexable-datatype", because I think "property-datatype" implies that it's the datatype as defined in the property definition. Also, use "datatype" throughout for consistency instead of "data-type".